### PR TITLE
Use ES2015 syntax for tests

### PR DESCRIPTION
**Pull request by SaschaNaz**
Looking at the tests I thought they should be rewritten in ES2015 before any possible fix here.

Additionally this removes lib-cov references that don't exist.

### CODE REVIEW

File: test/invalid.js [hunk 1]
```diff
@@ -1,42 +1,40 @@
...
+"use strict";
+
+const wp = require("../lib/webidl2");
+const expect = require("expect");
+const pth = require("path");
+const fs = require("fs");
+
+describe("Parses all of the invalid IDLs to check that they blow up correctly", () => {
+  const dir = pth.join(__dirname, "invalid/idl");
+  const skip = {};
```
Comment (marcoscaceres) [inline test/invalid.js]: the `skip` should probably be a `new Set()`. I'm not sure `skip` very useful tho, as all the names are going to be unique, right?

File: test/syntax.js [hunk 2]
```diff
@@ -1,41 +1,39 @@
...
+const pth = require("path");
+const fs = require("fs");
+const jdp = require("jsondiffpatch");
+const debug = true;
 
-    for (var i = 0, n = idls.length; i < n; i++) {
-        var idl = idls[i], json = jsons[i];
+describe("Parses all of the IDLs to produce the correct ASTs", () => {
+  const dir = pth.join(__dirname, "syntax/idl");
+  const skip = {}; // use if we have a broken test
```
Comment (marcoscaceres) [inline test/syntax.js]: as above, about `skip`.

File: test/invalid.js [hunk 3]
```diff
@@ -1,42 +1,40 @@
...
-                }
-                catch (e) {
-                    error = e;
-                }
-                finally {
-                    expect(error).toBeTruthy();
-                    expect(error.message).toEqual(err.message);
-                    expect(error.line).toEqual(err.line);
-                }
+  for (let i = 0, n = idls.length; i < n; i++) {
```
Comment (marcoscaceres) [inline test/invalid.js]: Nit: 
```
for(const idl of idl) {

}
```

Comment (marcoscaceres) [reply test/invalid.js]: fixed typo in code above

Comment (SaschaNaz) [reply test/invalid.js]: Couldn't do it because of `errors[i]`.

Comment (marcoscaceres) [reply test/invalid.js]: ah, ok

File: test/invalid.js (hunk 1 above)
Comment (SaschaNaz) [reply test/invalid.js]: Probably an array will be sufficient enough as `new Set()` will eventually be written as  `new Set(["testname1", "testname2"])`.

>I'm not sure skip very useful tho, as all the names are going to be unique, right?

I'm sorry but not sure why the uniqueness makes `skip` not useful, can you clarify more?

Comment (marcoscaceres) [reply test/invalid.js]: I guess the question is, under what circumstances will there be duplicate file names? (Or have I misunderstood what `skip` doing?) If the answer is "never", then the uniqueness check is not needed.

Comment (SaschaNaz) [reply test/invalid.js]: Well, I think `skip` is for temporarily disabling a test which cannot be fixed in the short term, or just a test-to-do-later.

Comment (marcoscaceres) [reply test/invalid.js]: I'm ok to keep it as is. If you are happy with it, I'm happy with it :)

Comment (SaschaNaz) [reply test/invalid.js]: I think the uniqueness is even an assumption for `skip` as it would accidentally skip multiple tests if names are not unique.

**Comment by marcoscaceres** [comment]
published as webidl2@6.0.1

**Pull request merged**